virt-v2v/pkg/customize/scripts/rhel/run/network_config_util.sh

```diff
@@ -181,25 +142,34 @@ udev_from_netplan() {
 
     # netplan with root dir
     netplan_get() {
-        if netplan_supports_get; then
-            netplan get --root-dir "$NETPLAN_DIR" "$@" 2>&3 
-        else
-            log 'Info: netplan not supporting get subcomment, using python'
-            netplan_get_py "$@" 2>&3
-        fi
+        netplan get --root-dir "$NETPLAN_DIR" "$@" 2>&3
     }
 
     # Loop over all interface names and treturn the one with target_ip, or null
     find_interface_by_ip() {
         target_ip="$1"
-
-        # Loop through all interfaces and check for the given IP address
-        netplan_get ethernets | grep -Eo "^[^[:space:]]+[^:]" | while read IFNAME; do
-            if netplan_get ethernets."$IFNAME".addresses | grep -q "$target_ip"; then
-                echo "$IFNAME"
+        if netplan_supports_get; then
+          # Loop through all interfaces and check for the given IP address
+          netplan_get ethernets | grep -Eo "^[^[:space:]]+[^:]" | while read IFNAME; do
+              if netplan_get ethernets."$IFNAME".addresses | grep -q "$target_ip"; then
+                  echo "$IFNAME"
+                  return
+              fi
+          done
+        else
+            netplan generate --root-dir "$NETPLAN_DIR"
+            NM_FILE=$(grep -El "Address[0-9]*=.*$S_IP.*$" "$SYSTEMD_NETWORK_DIR"/*)
```

Comment:
Let's add a check that the systemd_network_dir exists, just in case netplan is using a different backend